### PR TITLE
[docker,podman] Collect container image layers

### PR DESCRIPTION
**Pull request by TurboTurtle**
Adds collection of container image layers for the docker and podman plugins via the `history` and `tree` subcommands repsectively.

Resolves: #3796

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Review by jcastill — APPROVED** [review]
Tested with podman in Fedora and RHEL, works perfectly.

**Review by pmoravec — APPROVED** [review]

**Comment by samuelkarp** [comment]
We don't need to hold this open (we can do it in another PR), but we're interested in the image layer information from containerd as well.

**Comment by SergeyKanzhelev** [comment]
Maybe not resolve the ticket to wait for Containerd implementation? This PR says "Resolves: https://github.com/sosreport/sos/issues/3796" which will close the ticket once PR is merged

**Comment by TurboTurtle** [comment]
Dropped the `resolves`, was figuring we could get feedback for containerd during review but if we need to push that collection and these separately that's fine too.

**Comment by SergeyKanzhelev** [comment]
@samuelkarp does `ctr` supports something like this? What is the recommended way to collect this on Containerd?

**Comment by samuelkarp** [comment]
We don't really have a high-level interface for this in `ctr`.  Probably closest would be something like:

* `ctr images list` - grab all refs (first column) and digests (second column)
* `ctr content get <digest>` - for each digest
* For any digest that was then either an index or manifest list, grab the subsequent manifest digests from there and `ctr content get <digest>` again for the actual manifest
* If we _also_ want the config, then `ctr content get <digest>` on the config digest.  But configs can have private information in them (environment variables and command line, mostly).

**Pull request merged**
